index.js

```diff
@@ -2,6 +2,9 @@
  * Module dependencies
  */
 var assert = require('assert')
+var get = require('get-value')
+var set = require('set-value')
+var isObject = require('is-kindof').isObject;
```

Comment:
typeof is buggy, you know that. it will allow regexp, arguments, date and etc to pass the assertion. its always better to have some little lib for types instead of using instanceof/typeof